drivers: mbox: stm32-hsem: build driver if DT enabled

avolmat-st opened the pull request:
Default to building the driver if its node is enabled in the device-tree, avoiding the need to manually set CONFIG_MBOX_STM32_HSEM.

Commit:
drivers: mbox: stm32-hsem: build driver if DT enabled

Default to building the driver if its node is enabled in
the device-tree, avoiding the need to manually set
CONFIG_MBOX_STM32_HSEM.

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

erwango reviewed (changes requested):
Thanks but while you're at, please remove existing usage occurences in samples.

Commit:
samples: remove CONFIG_MBOX_STM32_HSEM enabling in conf files

Now that the MBOX_STM32_HSEM driver is enabled by default if
present within the device-tree, it is no more necessary to
manually enable it within samples board specific conf files.

Remove as well the CONFIG_LOG_BACKEND_UART since overall
CONFIG_LOG is anyway not enabled in the samples/drivers/mbox

Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>

mathieuchopstm commented on samples/drivers/mbox/remote/boards/stm32h747i_disco_stm32h747xx_m4.conf:
```diff
@@ -1,5 +1,2 @@
 CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
-CONFIG_LOG_BACKEND_UART=y
```
Erroneous removal?

avolmat-st replied:
No, in fact in samples/drivers/mbox there is no CONFIG_LOG enabled at all so this CONFIG doesn't make sense and just complain during the build. That's why I removed it.
I added a comment within the commit log to explain.
